src/devicemanager/devicemanager.h

```diff
@@ -30,7 +30,7 @@ class DeviceManager : public QAbstractListModel
      * @return DeviceManager*
      */
     static DeviceManager* self();
-    ~DeviceManager();
+    ~DeviceManager() override;
```

Comment:
The commit is not clear. Why is this necessary ? Why override is necessary ? Why are you overriding the desconstrutor. 
Is this overriding the parent deconstruction to avoid any problem related to the singleton nature ?